test/component_activated_form_dialog_test.js

```diff
@@ -0,0 +1,323 @@
+require('./setup');
+const expect = require('chai').expect;
+
+describe("ActivatedFormDialog", function() {
+
+  const ActivatedFormDialog = require('../app/javascript/src/component_activated_form_dialog');
+  const COMPONENT_CLASS_NAME = "ActivatedFormDialog";
+  const SUBMIT_BUTTON_ID = "submit-test-id";
+  const FORM_ID = "form-test-id";
+  const TEXT_CANCEL = "Dialog says Cancel";
+  var dialog;
+
+  before(function() {
+    // jQuyery is present in document because the
+    // components use it, so we can use it here.
+
+    var $form = $(`<form action="/" method="post">
+                       <input type="text" value="some value here" name="some_input" />
+                       <input type="submit" value="Submit button" />
+                     </form>`);
+
+    var $submit = $form.find("[type=submit]");
+
+    $submit.attr("id", SUBMIT_BUTTON_ID);
+    $form.attr("id", FORM_ID);
+    $(document.body).append($form);
+    dialog = new ActivatedFormDialog($form, {
+      cancelText: TEXT_CANCEL,
+      okText: $submit.val(),
+      activator: $submit
+    });
+  });
+
+  describe("HTML", function() {
+    it("should have the basic HTML in place", function() {
+      expect($("#" + FORM_ID).length).to.equal(1);
+    });
+
+    it("should have the component class name present", function() {
+      expect(dialog.$container.hasClass(COMPONENT_CLASS_NAME)).to.be.true;
+    });
+
+    it("should not have a two form buttons", function() {
```

Comment:
Is this line should be "should have a two form button"?

Reply:
Well spotted. I've just corrected.